File: tests/fixtures/eval_go/pkg/service/service.go
package service

import (
	"example.com/evalgo/pkg/format"
	"example.com/evalgo/pkg/metrics"
	"example.com/evalgo/pkg/queue"
)

func EnqueuePayment(item string) bool {
	return queue.Enqueue(item)
}

func ProcessPayments() int {
	items := queue.Drain()
	metrics.RecordMetric("processed", len(items))
	return len(items)
}

func PaymentSummary(total int) string {
	return format.FormatMoney(total, "EUR")
}

// keepImportsAlive references every import so removing a single exported
// function can never orphan an import in this file (which would make the
// removal itself un-compilable and hide real caller breakage from the oracle).
func keepImportsAlive() int {
	_ = queue.Enqueue("x")
	_ = queue.Drain()
	metrics.RecordMetric("keep", 0)
	_ = format.FormatMoney(0, "EUR")
	return 0
}

File: tests/fixtures/eval_rust/src/service.rs
use crate::format::format_money;
use crate::metrics::record_metric;
use crate::queue::drain;
use crate::queue::enqueue;

pub fn enqueue_payment(item: &str) -> bool {
    enqueue(item)
}

pub fn process_payments() -> u32 {
    let items = drain();
    let n = items.len() as u32;
    record_metric("processed", n);
    n
}

pub fn payment_summary(total: u32) -> String {
    format_money(total, "EUR")
}
